### PR TITLE
Add cdata method

### PR DESCRIPTION
**Pull request by adayton1**
For ManagedArray<T>, the data method always returns T*, and if T is not const the ManagedArray is always marked as touched. This pull request adds a cdata method that always returns const T* and never marks the ManagedArray as touched.

An alternative would be to make data a template function so that it could be used like `myArray.data<const int>`(). The implementation below is greatly simplified, but gives the general idea:
template <typename U>
U* data() { move(space, std::is_const<U>::value); return m_active_pointer; }

If you have better ideas, I will be happy to hear them.

### CODE REVIEW

File: src/chai/ManagedArray_thin.inl [hunk 1]
```diff
@@ -118,6 +118,12 @@ CHAI_HOST_DEVICE T* ManagedArray<T>::data() const
    return m_active_pointer;
 }
 
+template <typename T>
+CHAI_HOST_DEVICE T* ManagedArray<T>::cdata() const
```
[the remark this answers is not in the document]
Comment (adayton1) [reply src/chai/ManagedArray_thin.inl]: Absolutely. Good catch! Fixed.

File: src/chai/ManagedArray.inl [hunk 2]
```diff
@@ -499,6 +499,29 @@ T* ManagedArray<T>::data() const {
 #endif
 }
 
+template<typename T>
+CHAI_INLINE
+CHAI_HOST_DEVICE
+const T* ManagedArray<T>::cdata() const {
```
[the remark this answers is not in the document]
Comment (adayton1) [reply src/chai/ManagedArray.inl:505]: I'm not sure I understand the question. Could you clarify?

[the remark this answers is not in the document]
Comment (robinson96) [reply src/chai/ManagedArray.inl:505]: well, your confusion makes sense, I left out a critical word. I was wondering about issues if T is already a const, e.g. MA<const int>, and you end up returning a const const int.

[the remark this answers is not in the document]
Comment (adayton1) [reply src/chai/ManagedArray.inl:505]: It doesn't have any compiler warnings/errors for the const const case. I added some test cases to verify.

**Comment by davidbeckingsale** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**